core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -297,7 +299,14 @@ protected void validateAddedDataFiles(TableMetadata base, Long startingSnapshotI
    */
   protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startingSnapshotId,
                                                   Iterable<DataFile> dataFiles) {
-    validateNoNewDeletesForDataFiles(base, startingSnapshotId, null, dataFiles, true);
+    validateNoNewDeletesForDataFiles(base, startingSnapshotId, null, dataFiles, true,
+        sequenceNumberForNewDataFiles != null);
+  }
+
+  protected void validateNoNewDeletesForDataFiles(TableMetadata base, Long startingSnapshotId,
```

Comment:
Could you copy the javadoc for the previous version? I think it's helpful to have it.

Reply:
added back